ctest-next/src/template.rs

```diff
@@ -2,78 +2,185 @@ use askama::Template;
+#[derive(Clone, Debug, Default)]
+pub(crate) struct TestTemplate {
+    pub(crate) const_cstr_tests: Vec<TestCstr>,
+    pub(crate) const_tests: Vec<TestConst>,
+    pub(crate) test_idents: Vec<BoxStr>,
+}
+
+impl TestTemplate {
+    pub(crate) fn new(
+        ffi_items: &FfiItems,
+        generator: &TestGenerator,
+    ) -> Result<Self, TranslationError> {
+        let helper = TranslateHelper {
             ffi_items,
-            translator: Translator::new(),
             generator,
+            translator: Translator::new(),
+        };
+
+        /* Figure out which tests are to be generated. */
+        // FIXME(ctest): Populate more test information, maybe extract into separate methods.
+        // The workflow would be to create a struct that stores information for the new test,
+        // and populating that struct here, so that the also things that have to be added to
+        // the test templates are the new tests parameterized by that struct.
+
+        let mut const_tests = vec![];
+        let mut const_cstr_tests = vec![];
+        for constant in ffi_items.constants() {
+            if let syn::Type::Ptr(ptr) = &constant.ty {
+                let is_const_c_char_ptr = matches!(
+                    &*ptr.elem,
+                    syn::Type::Path(path)
+                        if path.path.segments.last().unwrap().ident == "c_char"
+                        && ptr.mutability.is_none()
+                );
+                if is_const_c_char_ptr {
+                    let item = TestCstr {
+                        test_ident: cstr_test_ident(constant.ident()),
+                        rust_ident: constant.ident().into(),
+                        c_ident: helper.c_ident(constant).into(),
+                        c_type: helper.c_type(constant)?.into(),
+                    };
+                    const_cstr_tests.push(item)
+                }
+            } else {
+                let item = TestConst {
+                    test_ident: const_test_ident(constant.ident()),
+                    rust_ident: constant.ident.clone(),
+                    rust_type: constant.ty.to_token_stream().to_string().into_boxed_str(),
+                    c_ident: helper.c_ident(constant).into(),
+                    c_type: helper.c_type(constant)?.into(),
+                };
+                const_tests.push(item)
+            }
         }
+
+        let mut test_idents = vec![];
+        test_idents.extend(const_cstr_tests.iter().map(|test| test.test_ident.clone()));
+        test_idents.extend(const_tests.iter().map(|test| test.test_ident.clone()));
+
+        Ok(Self {
+            const_cstr_tests,
+            const_tests,
+            test_idents,
+        })
     }
+}
+
+/// Information required to test a constant CStr.
+#[derive(Clone, Debug)]
+pub(crate) struct TestCstr {
+    pub(crate) test_ident: BoxStr,
+    pub(crate) rust_ident: BoxStr,
+    pub(crate) c_ident: BoxStr,
+    pub(crate) c_type: BoxStr,
```

Comment:
```suggestion
    pub test_ident: BoxStr,
    pub rust_ident: BoxStr,
    pub c_ident: BoxStr,
    pub c_type: BoxStr,
```

Optional nit: if a struct isn't public, `pub` (without crate) is fine to get the same visibility as the struct itself